src/mpi/coll/transports/generic/transport.c

```diff
@@ -0,0 +1,1243 @@
+                           sched->tag, vtxp->nbargs.recv_reduce.comm->mpid_comm,
+                           &vtxp->mpid_req[0]);
+                MPIR_Grequest_start_impl(MPIR_COLL_GENERIC_queryfcn, NULL, NULL, &vtxp->nbargs.recv_reduce,
+                                         &vtxp->mpid_req[1]);
+
+                MPL_DBG_MSG_FMT(MPIR_DBG_COLL,VERBOSE,(MPL_DBG_FDEST,"  --> GENERIC transport (recv_reduce) issued\n"));
+
+                /* record vertex issue */
+                MPIR_COLL_GENERIC_record_vtx_issue(vtxp, sched);
+            }
+            break;
+
+        case MPIR_COLL_GENERIC_KIND_REDUCE_LOCAL:
+            if (vtxp->nbargs.reduce_local.flags == -1 ||
+                vtxp->nbargs.reduce_local.flags & MPIR_COLL_FLAG_REDUCE_L) {
+                MPL_DBG_MSG_FMT(MPIR_DBG_COLL,VERBOSE,(MPL_DBG_FDEST,"Left reduction vtxp->nbargs.reduce_local.flags %ld \n",
+                         vtxp->nbargs.reduce_local.flags));
+                MPIR_Reduce_local((const void *) vtxp->nbargs.reduce_local.inbuf,
+                                       (void *) vtxp->nbargs.reduce_local.inoutbuf,
+                                       vtxp->nbargs.reduce_local.count,
+                                       vtxp->nbargs.reduce_local.dt, vtxp->nbargs.reduce_local.op);
+                MPL_DBG_MSG_FMT(MPIR_DBG_COLL,VERBOSE,(MPL_DBG_FDEST,"  --> GENERIC transport (reduce local_L) complete\n"));
+            } else {
+                MPL_DBG_MSG_FMT(MPIR_DBG_COLL,VERBOSE,(MPL_DBG_FDEST,"Right reduction vtxp->nbargs.reduce_local.flags %ld \n",
+                         vtxp->nbargs.reduce_local.flags));
+                /* An extra data copy is required to do the R reduction */
+                MPIR_Reduce_local((const void *) vtxp->nbargs.reduce_local.inoutbuf,
+                                       (void *) vtxp->nbargs.reduce_local.inbuf,
+                                       vtxp->nbargs.reduce_local.count,
+                                       vtxp->nbargs.reduce_local.dt, vtxp->nbargs.reduce_local.op);
+
+                MPIR_Localcopy(vtxp->nbargs.reduce_local.inbuf,
+                               vtxp->nbargs.reduce_local.count,
+                               vtxp->nbargs.reduce_local.dt,
+                               vtxp->nbargs.reduce_local.inoutbuf,
+                               vtxp->nbargs.reduce_local.count, vtxp->nbargs.reduce_local.dt);
+                MPL_DBG_MSG_FMT(MPIR_DBG_COLL,VERBOSE,(MPL_DBG_FDEST,"  --> GENERIC transport (reduce local_R) complete\n"));
+            }
+
+            /* record vertex completion */
+            MPIR_COLL_GENERIC_record_vtx_completion(vtxp, sched);
+            break;
+        }
+    }
+
+    MPIR_FUNC_VERBOSE_EXIT(MPID_STATE_MPIR_COLL_GENERIC_ISSUE_VTX);
+}
+
+#undef FUNCNAME
+#define FUNCNAME MPIR_COLL_GENERIC_poke
+/* Transport function to make progress on the schedule */
+int MPIR_COLL_GENERIC_poke(MPIR_COLL_GENERIC_sched_t * sched)
```

Comment:
This function should return an error code and the value of whether or not the schedule is done should be an `int *` input argument.

Reply:
Ok.

Reply:
Done